### PR TITLE
unit test for go/sets/set.go

### PR DESCRIPTION
**Pull request by Its-Maniaco**
Partial Fix to #14931 

Covers unit test for `go/sets/set.go`

image

`ok      vitess.io/vitess/go/sets        0.011s  coverage: 73.8% of statements`

### CODE REVIEW

File: go/sets/set_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package sets
+
+import (
+	"testing"
+
```
Comment (frouioui) [inline go/sets/set_test.go:5]: license header is missing

Comment (Its-Maniaco) [reply go/sets/set_test.go:21]: Added.

**Comment by Its-Maniaco** [comment]
I can edit the comments and add further test if this looks fine.

File: go/sets/set_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (dbussink) [inline go/sets/set_test.go:2]: It's 2024 :smile:.

**Review by frouioui — APPROVED** [review]

File: go/sets/set_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	testSet.Insert(4, 5, 6)
+	compareSet := New[int](1, 2, 3, 4, 5, 6)
+	assert.Equal(t, testSet, compareSet)
+
+	assert.True(t, testSet.Equal(compareSet))
+	compareSet.Insert(-1, -2)
+	assert.False(t, testSet.Equal(compareSet))
+
+	//tests for Difference func
```
[the remark this answers is not in the document]
Comment (Its-Maniaco) [reply go/sets/set_test.go:38]: I bundled all of them since all of the functions were operations being done on a Set.

**Comment by dbussink** [comment]
@Its-Maniaco Looks like the `DCO` check is failing, can you make sure the commits have the correct signed-off messages and update the PR?

**Review by GuptaManan100 — COMMENTED** [review]
The changes look good to me now! What is the test coverage for this package post the changes?

**Comment by Its-Maniaco** [comment]
`ok      vitess.io/vitess/go/sets        (cached)        coverage: 78.6% of statements`

**Comment by GuptaManan100** [comment]
@Its-Maniaco I have pushed a commit to make the test coverage 100%. This should be good to be merged now.

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**